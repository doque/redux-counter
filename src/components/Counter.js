import React from "react";

// <Counter count={ 3 }Â />
const Counter = ({ count }) =>
  <div>
    {count}
  </div>;

export default Counter;
