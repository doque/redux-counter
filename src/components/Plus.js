import React from "react";
import Button from "./styled/Button";

const Plus = ({ increment }) =>
  <Button up onClick={increment}>
    +
  </Button>;

export default Plus;
