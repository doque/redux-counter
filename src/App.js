import React from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { increment, decrement } from "./actions/counterActions";

import "./App.css";
import Counter from "./components/Counter";
import Plus from "./components/Plus";
import Minus from "./components/Minus";

const App = ({ count, decrement, increment }) =>
  <div className="App">
    <Minus decrement={decrement} />
    <Counter count={count} />
    <Plus increment={increment} />
  </div>;

const mapStateToProps = state => ({
  count: state.count
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({ decrement, increment }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(App);

// For Testing, export unconnected Component
export { App };
