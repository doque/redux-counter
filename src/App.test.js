import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import { App } from "./App";
import Plus from "./components/Plus";
import Counter from "./components/Counter";

describe("App Component", () => {
  let wrapper;
  // this resets our component to a clean state before each test
  // ensuring that tests are not interefering with each other
  beforeEach(() => {
    wrapper = mount(<App count={0} />);
  });

  it("handles a click on Plus", () => {
    const spy = jest.fn();
    wrapper.setProps({
      increment: spy
    });
    wrapper.find(Plus).simulate("click");
    wrapper.setProps({ count: 20 });
    expect(spy).toBeCalled();
  });

  it("renders all sub components correctly", () => {
    const tree = renderer.create(<App count={31} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
