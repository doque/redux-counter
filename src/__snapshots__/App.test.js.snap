// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders all sub components correctly 1`] = `
<div
  className="App"
>
  <button
    onClick={undefined}
  >
    -
  </button>
  <div>
    Traalala 
    31
  </div>
  <button
    onClick={undefined}
  >
    +
  </button>
</div>
`;
